plutus-pab/README.md

```diff
@@ -0,0 +1,331 @@
+# Plutus Application Backend
+
+_PAB_ (Plutus Application Backend) is an Off-Chain application for managing the state of
+Plutus contract instances.
+
+## Table of Contents
+
+- [Building](#building)
+- [Running](#running)
+- [PAB Components](#pab-components)
+
+## Building
+
+_PAB_ is a Cabal based Haskell project. Production builds are done with Nix using [haskell.nix](https://github.com/input-output-hk/haskell.nix)
+
+### Cabal
+
+```
+$ cabal build
+```
+
+### Nix
+
+```
+$ nix-build ../default.nix -A plutus.haskell.packages.plutus-pab
+```
+
+## Running
+
+In order to use PAB several of its components need to be started. Furthermore the pab-client has
+to be started in order to access the web frontend. The required steps are described below.
+
+First we build the startup scripts:
+
+```
+$ nix-build ../default.nix plutus-pab.demo-scripts
+```
+
+Next we start all required servers and install several contracts:
+
+```
+$ ./result/bin/pab-start-all-servers
+```
+
+Now we start an additional node:
+
+```
+$ ./result/bin/pab-start-second-pab
+```
+
+Finally we can start the pab web frontend:
+
+```
+$ cd ../plutus-pab-client
+$ npm start
+```
+
+The client is now running at https://localhost:8009 -- See [pab-demo-scripts.nix](https://github.com/input-output-hk/plutus/blob/pab-readme/plutus-pab-client/pab-demo-scripts.nix) for details on the service invcation and contract installation.
+
+## PAB Components
+PAB contains several commands and services, which are outlined below.
+
+- [psgenerator](#psgenerator)
+- [migrate](#migrate)
+- [all-servers](#all-servers)
+- [client-services](#client-services)
+- [wallet-server](#wallet-server)
+- [webserver](#webserver)
+- [node-server](#node-server)
+- [chain-index](#chain-index)
+- [metadata-server](#metadata-server)
+- [signing-process](#signing-process)
+- [default-logging-config](#default-logging-config)
+- [process-outboxes](#process-outboxes)
+
+### psgenerator
+
+```
+$ cabal run plutus-pab -- psgenerator
+```
+
+#### Description
+
+Generates the purescript bridge code.
+
+#### Source
+
+- [app/PSGenerator.hs](https://github.com/input-output-hk/plutus/blob/master/plutus-pab/app/PSGenerator.hs)
+
+### migrate
+
+```
+$ cabal run plutus-pab -- migrate
+```
+
+#### Description
+
+Migrates the database in `pab-core.db` to the current schema.  The database contains the state for the eventful library. Please refer to the  [eventful docs](https://hackage.haskell.org/package/eventful-sqlite-0.2.0/docs/Eventful-Store-Sqlite.html) for more information.
+
+#### Source
+[Plutus.PAB.App.migrate](https://github.com/input-output-hk/plutus/blob/master/plutus-pab/src/Plutus/PAB/App.hs#L283)
+
+
+#### all-servers
+
+```
+$ cabal run plutus-pab -- all-servers
```

Comment:
Does it work with the config yaml argument? (maybe there's a default value but I can't remember)

Reply:
I added a comment below. It would be nice to later have a less manual way of doing this but that's outside of scope for this PR - i will tackle that at a later point.